apis/tags/v1beta1/tagbinding_identity.go

```diff
@@ -13,3 +13,107 @@
 // limitations under the License.
 
 package v1beta1
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/apis/common/identity"
+	refsv1beta1 "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ identity.Identity = &TagBindingIdentity{}
+
+const TagBindingIDURL = "tagBindings/{{parentWithFullURL}}/tagValues/{{tagValueID}"
+
+type TagBindingIdentity struct {
+	// parent is the full resource name of the resource to which the tag is bound.
+	// e.g. `//cloudresourcemanager.googleapis.com/projects/12345`
+	parent string
+	// tagValue is the resource name of the TagValue.
+	tagValue string
+}
+
+func (i *TagBindingIdentity) String() string {
+	// The name of the TagBinding. This is a String of the form:
+	// `tagBindings/{full-resource-name}/{tag-value-name}`
+	// (e.g. `tagBindings/%2F%2Fcloudresourcemanager.googleapis.com%2Fprojects%2F123/tagValues/456`).
+	// The parent needs to be URL-encoded.
+	return "tagBindings/" + url.PathEscape(i.parent) + "/tagValues/" + i.tagValue
+}
+
+func (i *TagBindingIdentity) ParentWithFullURL() string {
+	return i.parent
+}
+
+func (i *TagBindingIdentity) TagValue() string {
+	return "tagValues/" + i.tagValue
+}
+
+func (i *TagBindingIdentity) FromExternal(ref string) error {
```

Comment:
tagBinding ID is not named. It's a 1:1 mapping between <resource> and TagValue. Thus, the ID  is in the form of "tagBindings/<<parent-full-url-with-url-escaped>>/tagValues/{{tagValueID}}"